### ISSUE TITLE
Sync read page progress with trackers

### ISSUE THREAD

**Issue by kanjieater** [mihonapp/mihon#236]
### Describe your suggested feature

Currently the behavior of only syncing when a chapter is read is limiting. If mihon could sync per page, you could read something on your phone, then pick up where you left off on a desktop or other device.

There has been interest voiced for this with Komga, suwayomi, to name a few.

Original issue was here https://github.com/tachiyomiorg/tachiyomi/issues/5978

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.16.1](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by barrelltitor** [comment]
This would be a great addition. Happens quite often I might be reading a comic on one device, then I want to finish it on another device

**Comment by RandomNamer** [comment]
I don't think trackers are meant for syncing read progress across devices at such granularity. That said, I tried to implement a workable version of [page-based Komga tracker](https://github.com/RandomNamer/mihon/commit/0a841188be939d4fddb6c3d8b713bba80f3f680c) for myself. It's not a elegant solution though: to sync you have to click tracker icon, wait for a few seconds before close the dialog for the request to be done and page progress writen to the DB.

**Comment by RandomNamer** [comment]
[app-dev-arm64-v8a-preview.apk.zip](https://github.com/user-attachments/files/16279777/app-dev-arm64-v8a-preview.apk.zip)
> making a reverse proxy for it to high jack on syncs

For Komga it would be easier to just send read-progress requests in app to achieve sync. In this implementation, you can see the page progress get updated in Komga web app each time you turned page in Mihon, cuz it uses a event stream to get updates from server in real time. However to sync server progress to Mihon you need to manually trigger tracker refreshing.

**Comment by kanjieater** [comment]
It's working great so far. Thank you so much for sharing.

**Comment by garret** [comment]
I just installed your Mihon apk but to me the tracking does not work. Should I do anything in particular? I am reading a manga from Komga. when I try to enable "Tracking" I have the "Update progress after reading" enabled as well as the Komga 'enhanced tracker'. However, it seems no page progress is synced to Komga :/

**Comment by garret** [comment]
Update: I managed to get the syncing but it seems only one way: only mihon sends page reading progress to komga but if I advance reading on komga the progress is not synced back to mihon. Is this the supposed behavior?

**Comment by kanjieater** [comment]
> However to sync server progress to Mihon you need to manually trigger tracker refreshing.

To do this, click the tracker button for the comic, it should show the kkmga sync progress which then syncs mihon with the komga seever in the background

**Comment by garret** [comment]
@kanjieater yes I had managed the syncing but I wrote a later reply where I got the syncing work but it seems only one way: Mihon sends page progress to komga web interface but if I progress a comic from the web interface that is not synced back in Mihon. But I don't know if that is the expected behavior?

**Comment by kanjieater** [comment]
> @kanjieater yes I had managed the syncing but I wrote a later reply where I got the syncing work but it seems only one way: Mihon sends page progress to komga web interface but if I progress a comic from the web interface that is not synced back in Mihon. But I don't know if that is the expected behavior?

No - im telling you you have to click the tracker manually each time to get komga to sync to mihon.

**Comment by garret** [comment]
Thank you it worked. I really had to spend some time in understanding this tracker button to tap because it doesn't look like a button at all. In an ideal world, I would have liked that there should be no need to "tap to sync" but I understand that at the moment this is the best that it is. Thank you for the APK and I hope your PR gets pushed through!

**Comment by UtterVitriol** [comment]
I have been using komelia over mihon simply because it is able to sync at the page level with komga. I do prefer mihon though, so I hope this gets implemented eventually.
